wifi-subsys/components/uart/src/at_handler.c

```diff
@@ -0,0 +1,213 @@
+};
+
+char *symbols[4] = {
+    "+",
+    "&",
+    "?",
+    "=",
+};
+
+int get_symbol(char *command, size_t len, char *output) {
+    for (size_t i = 0; i < len; i++) {
+        for (size_t j = 0; j < 4; j++) {
+            if (command[i] == symbols[j][0]) {
+                memcpy(output, symbols[j], 2);
+                return ESP_OK;
+            }
+        }
+    }
+    return ESP_FAIL;
+}
+
+esp_err_t parse_at_message(uint8_t *at_comant, at_request_t *output) {
+    size_t at_size = strlen((char *)at_comant) + 1;
+    char commands[at_size];
+    memcpy(commands, at_comant, at_size);
+    char plus[2];
+    memset(plus, "0", sizeof(plus));
+    get_symbol(commands, at_size, plus);
+    char *key = strtok(commands, plus);
+    char firts_step[100];
+    char *key_value = NULL;
+    char *value = NULL;
+
+    int i = 0;
+    char second_character[2];
+    while (key != NULL) {
+        if (i == 1) {
+            memcpy(firts_step, key, sizeof(firts_step));
+        }
+        i++;
+
+        key = strtok(NULL, plus);
+    }
+
+    get_symbol(firts_step, at_size, second_character);
+    char *key2 = strtok(firts_step, second_character);
+    i = 0;
+    while (key2 != NULL) {
+        if (i == 0) {
+            key_value = (char *)malloc(strlen(key2) + 1);
+
+            memset(key_value, 0, strlen(key2) + 1);
+            memcpy(key_value, key2, strlen(key2) + 1);
+        }
+        if (i == 1) {
+            value = (char *)malloc(strlen(key2) + 1);
+            memcpy(value, key2, strlen(key2) + 1);
+        }
+        i++;
+        key2 = strtok(NULL, "+");
+    }
+
+    /* cppcheck-suppress nullPointerRedundantCheck
+     * (reason: <is necessary verify that this data is not null>) */
+    memcpy(output->key, key_value, strlen(key_value) + 1);
+    if (value != NULL) {
+        memcpy(output->value, value, strlen(value) + 1);
+    }
+
+    memcpy(output->action, plus, 2);
+    memcpy(output->read_or_write, second_character, 2);
+
+    if (key_value == NULL && value == NULL) {
+        return ESP_FAIL;
+    }
+
+    free(value);
+    free(key_value);
+
+    return ESP_OK;
+}
+
+int at_cmd_to_name(char *key) {
+    for (size_t i = 0; i < ARRAY_SIZE; i++) {
+        key[strlen(key)] = 0;
+        if (strcmp(key, default_keys[i]) == 0) {
+            return i;
+        }
+    }
+
+    return -1;
+}
+
+void action_request(enum at_keys_n at_key, at_request_t *payload) {
+    esp_err_t err = ESP_FAIL;
+    switch (at_key) {
+    case (NVS_RST):
+        err = wifi_restore_default();
+        break;
+    case (WIFI_OFF):
+        err = wifi_off();
+        break;
+    case (WIFI_ON):
+        err = wifi_on();
+        break;
+    case (WIFI_RST):
+        err = wifi_restart();
+        break;
+
+    default:
+        ESP_LOGE(__func__, "unhandler event %i \n", at_key);
+        err = ESP_FAIL;
+        break;
+    }
+
+    if (err != ESP_OK) {
+        sendData(__func__, "\r\n 400 \r\n");
```

Comment:
Why the spaces before and after _"400"_ ?
```suggestion
        sendData(__func__, "\r\n400\r\n");
```